src/lib/components/gallery-main.jsx

```diff
@@ -18,8 +18,13 @@ export const GalleryMain = ({ renderGalleryItem, className, ...props }) => {
     touchState,
     setTouchState,
     swipeThreshold,
+    goToIndex,
   } = useGallery()
 
+  useEffect(() => {
+    if (!galleryItems[activeIndex]) goToIndex(0)
```

Comment:
I think we can lose the `useEffect`, and integrate this logic into the actual `goToIndex` function. That thing should be updated to be a bit more defensive anyway. Inside that function, maybe instead of just accepting that `index` is defined, let's first check that it's valid, and if not, set to zero.

I think with this useEffect solution, we'll still briefly get a warning, since side-effects run _after_ the render phase.

Reply:
Agree with this one, otherwise we'll always have to resort to this `useEffect` trick.